com.unity.multiplayer.mlapi/Runtime/Core/NetworkUpdateLoop.cs

```diff
@@ -237,6 +240,16 @@ public static PlayerLoopSystem CreateLoopSystem()
         [RuntimeInitializeOnLoadMethod]
         private static void Initialize()
         {
+#if UNITY_EDITOR
+            EditorApplication.playModeStateChanged += stateChange =>
+            {
+                if (stateChange == PlayModeStateChange.ExitingPlayMode)
+                {
+                    PlayerLoop.SetPlayerLoop(PlayerLoop.GetDefaultPlayerLoop());
```

Comment:
This doesn't feel right. We start with CurrentPlayerLoop, not DefaultPlayerLoop, so feels odd overwriting any other customizations with the default loop.

Reply:
> Why do you say NetworkUpdateLoop will still tick systems after we stop?

DomainReload happens when we enter PlayMode and that'd cause `NUL` to be injected into `PlayerLoop`.
However, when we exit PlayMode (stop playing), nothing reloads so that `PlayerLoop` still executes its systems and `NetworkUpdateLoop` systems are still being ticked therefore `NUL` still ticks `INetworkUpdateSystem` instances underneath.
As I mentioned in the PR description, the proper handling of this scenario is to unregister from `NUL` on `OnApplicationQuit`, `OnDestroy`, `OnDisable` but sometimes that's not the case, people might forget and/or somehow unregister codepath might not be executed.

> If this is supposed to be a fix for Noel's #558 / MTT-529 can you please note that in the PR description or title somewhere.

No, we're just playing safe here and revert `PlayerLoop` changes back when we exit PlayMode in the Editor.
This is not a fix for #558 — this is just hardening.

> why are our systems still running but systems like behaviourupdate, audio, input etc. aren't having issues?

I can't paste codes here for obvious reasons :) but I can say that they're being protected if not playing. That's not a `PlayerLoop` logic, that's systems' own logic to execute or skip. `PlayerLoop` is always around and being ticked, it never stops. Systems under `PlayerLoop` might pause/stop/start or do whatever they want. They're being registered into low-level engine ticking loop, in this case `NetworkUpdateLoop` is still being ticked by the `PlayerLoop` — that's completely normal.

> This doesn't feel right. We start with CurrentPlayerLoop, not DefaultPlayerLoop, so feels odd overwriting any other customizations with the default loop.

We do that when we stop playing in the Editor (exit PlayMode), not when we enter PlayMode.
(We start with the default playerloop after domainreload anyway, systems relying on playerloop injection should already re-register after domainreload anyway).
I do agree that it might be more granular and take-out only the `NUL`-injected systems from the `PlayerLoop` — I'll do that, it'd be even safer and better approach IMO.